File: src/pages/exercises/_string-table-generator/_codes/js/response/table-generator.js
function createTable(rows, cols, text) {
  let result = '';
  text = text ? `${text} ` : '';

  for (let row = 1; row <= rows; row++) {
    for (let col = 1; col <= cols; col++) {
      result += `${text}${row}.${col}`;
      result += col === cols ? '\n' : ' ';
    }
  }

  return result;
}

function createTableHTML(rows, cols, text) {
  let result = '<table>\n';
  text = text ? `${text} ` : '';

  for (let row = 1; row <= rows; row++) {
    result += '  <tr>\n';
    let tds = [];
    for (let col = 1; col <= cols; col++) {
      tds.push(`    <td>${text}${row}.${col}</td>\n`);
    }
    result += tds.join('');
    result += '  </tr>\n';
  }

  result += '</table>\n';
  return result;
}

export { createTable, createTableHTML };

File: src/pages/problems/_string-table-generator/_codes/ruby/response/src/table_generator.rb
class TableGenerator
  def TableGenerator.create_table(rows, cols, text = '')
    result = '';
    text = text != '' ? "#{text} " : ''
    for row in (1..rows)
      for col in (1..cols)
        result += "#{text}#{row}.#{col}"
        result += col == cols ? "\n" : ' '
      end
    end 
    return result
  end

  def TableGenerator.create_table_HTML(rows, cols, text = '')
    result = "<table>\n";
    text = text != '' ? "#{text} " : ''
    for row in (1..rows)
      result += "  <tr>\n";
      tds = []
      for col in (1..cols)
        tds << "    <td>#{text}#{row}.#{col}</td>\n"
      end
      result += tds.join('')
      result += "  </tr>\n"
    end
    result += "</table>\n"   
  end
end
